front/src/utils/rtk.ts

```diff
@@ -0,0 +1,33 @@
+import type {
+  MutationDefinition,
+  OverrideResultType,
+  QueryDefinition,
+} from '@reduxjs/toolkit/query';
+
+export type OverrideQueryArgType<Definition, NewQueryArgType> =
```

Comment:
This handles both queries and mutation, so we could leave out `Query` from the name (would be more consistent with the existing `OverrideResultType` helper from upstream).